src/main/java/cohen/maze/Maze.java

```diff
@@ -0,0 +1,166 @@
+package cohen.maze;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Random;
+
+public class Maze {
+    private Cell[][] maze;
+    private int height;
+    private int width;
+
+    //generate the full 2d array of cell objects. all the 'visited's are false and all the walls are true(that's done in the cells constructor).
+    public Maze(int height, int width) {
+        //initialize the number of rows and columns in the maze with the numbers entered
+        this.height = height;
+        this.width = width;
+
+        //initialize the 2d maze with the number of spaces allocated by the height and width entered
+        maze = new Cell[height][width];
+        //fill the maze with cells
+        generateMaze();
+        //dig a an open path through the maze
+        generatePath();
+        //print the maze in the console
+        printMaze();
+    }
+
+    /**
+     * The generateMaze method fills the maze with unvisited, walled cells.
+     */
+    public void generateMaze() {
+        for(int i = 0; i < height; i++) {
+            for(int j = 0; j < width; j++) {
+                maze[i][j] = new Cell(i, j);
+            }
+        }
+    }
+
+    /**
+     * The generatePath method digs or generates a random open path through the maze of cells.
+     */
+    private void generatePath() {
+        //Find the starting point to start digging.
+        Random x = new Random();
+        int row = x.nextInt(height);
+        Random y = new Random();
+        int col = y.nextInt(width);
+
+        //starting cell
+        maze[row][col].setVisited(true);
+
+        //start digging the path
+
+        dig(row, col);
+
+
+
+    }
+
+    //Depth-First Searching Algorithm that digs in increments of 1
+    public void dig(int row, int column){
+        //generate four random directions
+        Integer[] randDirs = generateRandomDirections();
+        //examine each direction
+        for(int i = 0; i < randDirs.length;i++) {
+            switch (randDirs[i]) {
+                case 1: //Up
```

Comment:
make 1, 2, 3, 4 constants

Reply:
In what way? Should I make the arraylist constant and filled with 4 constants? Should I make them in the beginning, or in the generateRandomDirections() method?

Reply:
Or do you mean just the case numbers?

Reply:
Also, why?

Reply:
Because NORTH, SOUTH, EAST, WEST is a lot nicer to read than 1,2,3,4

Reply:
done